### ISSUE TITLE
In Firefox form submission fails when subscribed to web push

### ISSUE THREAD

**Issue by miczh** [OneSignal/OneSignal-Website-SDK#247]
In Firefox, when using input type="file" in html forms (method="POST" enctype="multipart/form-data"), submission is not working anymore, when subscribed to web push. Forms without type="file" still work. In chrome, there aren't any problems.

Please try if you get the same error in firefox:

1. try uploading a file (without being subscribed for web push) -> works
2. try uploading a file again after subscribing for web push -> error

**Comment by jasonpang** [comment]
Hey @miczh,

Do you happen to have a page we can demo with this issue?

**Comment by miczh** [comment]
Hey Jason
You can check [here](https://push.zuestmedia.ch/fftest.php)

**Comment by jasonpang** [comment]
Hey @miczh,

I'm not seeing any network or console errors from uploading a file after subscribing. I've tested that like this: https://giphy.com/gifs/xT39CT2kUfJEQ41VAI/fullscreen. Are your testing steps any different?

Can you describe the error you're getting or upload a screenshot of it?

**Comment by miczh** [comment]
Hey @jasonpang 

I tested on my old mac as well now and there everything works in ff ;-). On my windows 10 machine it isn't working. 

it doesn't result in an console / network error, the form submission starts, but doesn't finish and runs for minutes...

**Comment by jasonpang** [comment]
Hey @miczh,

This probably isn't a web SDK issue, at least I can't think of anything specific that would cause a post request to go through. We're no longer polyfilling browser APIs related to network requests, so everything should be just fine.

You can check chrome://net-internals to maybe see what's going on with that request.

**Comment by miczh** [comment]
Hey @jasonpang 

You are right. I tested again subscribed to notifications, but without the onesignal scripts loaded. The result stays as described above. after submitting the form, it runs into timeout without happening anything. Therefore it's probably a firefox bug...

**Comment by jasonpang** [comment]
@miczh,

Is a service worker installed? You can check about:debugging or about:serviceworkers to see if one is installed. It can persist even if OneSignal isn't loaded as a script, if you previously subscribed. Are you using an old version of Firefox? Version 44 had a bug where some fetch requests wouldn't go through due to the browser omitting some required headers if a service worker added a fetch handler. If unregistering the service worker fixes the issue, there might be some bug with the service worker and we'd like to explore the issue further.

If you see some kind of relation with the bug and the service worker and you're still not sure what the issue is, can you schedule a time to demo the issue here https://calendly.com/jasonpang/onesignal so that we can investigate it together?

**Comment by miczh** [comment]
Hey @jasonpang 

It has probably something to do with the service worker. 

after dismissing the loaded service worker from onesignal, with about:serviceworkers in ff (actual version 55.0.1) the submission worked again.

**Comment by jasonpang** [comment]
@miczh,

Can you provide an updated test URL one more time?

**Comment by miczh** [comment]
Hey @jasonpang 

[link1 ](https://push.zuestmedia.ch/fftest.php)is with onesignal script loaded
[link2 ](https://push.zuestmedia.ch/fftest2.php)is without onesignal script loaded

sorry, i deleted it after last test before service worker hint ;-)

**Comment by jasonpang** [comment]
Just retested in Firefox 54.0.1 on Mac OS X. I'm not able to reproduce the issue and both link1 and link2 work. That is, both with and without the service worker, the form upload succeeds for me. Checking the Network panel, the request shows a 200 OK response for each test without any timeout (both tests are very quick and pass in less than a second).

I'm afraid I'm not sure where you can look to find out more information. Maybe you might see some errors in the browser console, or you might have to dig deeper and find a way to trace the complete network activity.

![image](https://user-images.githubusercontent.com/1015970/29334968-0803f45e-81be-11e7-8d64-7d2edc4b384b.png)

**Comment by miczh** [comment]
Hi @jasonpang 

With MacOS and firefox i'm also not getting errors. It is just on Windows (10 pro) and firefox (ver. 55.0.1). 

But there is no error message or console log, nothing, just waiting for timeout. after like 300 seconds...

This problem is not that urgent to me, i am using usually chrome :-). And if you can't replicate it, it has possibly to do with my system. 

Did you test it on a windows machine as well?

**Comment by jasonpang** [comment]
Hey @miczh,

We tested on Windows and we were able to reproduce the timeout error. It looks like it happens if a service worker is registered (it doesn't even have to be started) and even if the service worker doesn't have a "fetch handler" (which is something that allows the service worker to intercept network requests). I tried enabling network logging, but I didn't see anything relevant. It sounds like a Firefox bug, and I'll be reporting this bug later (and post a link here).
